11.LoDSystem/main.cpp

```diff
@@ -686,21 +704,23 @@ class LoDSystemApp : public ApplicationBase
                         auto& drawCallOffsetsInDWORDs = lodLibraryData.drawCallOffsetsIn20ByteStrides;
                         for (auto i = 0u; i < cullingParams.drawcallCount; i++)
                             drawCallOffsetsInDWORDs[i] = lodLibraryData.drawCallOffsetsIn20ByteStrides[i] * sizeof(asset::DrawElementsIndirectCommand_t) / sizeof(uint32_t);
+                        video::IGPUBuffer::SCreationParams p1; p1.size = cullingParams.drawcallCount * sizeof(uint32_t);
```

Comment:
You'll definitely run into a validation error because you need to set `usage` for this creation params, default is `EUF_NONE` which is invalid to pass to buffer creation.

Well how can I figure out the usage Erfan?!
1. Do it like a normal human being: Investigate how these buffers are used in this example
2. Do It like a Chad: set it to some arbitrary Usage and let the vulkan validation errors tell you what you missed. (for example vulkan will say, `Hey fucker you used this buffer for that reason but didn't pass "VK_BUFFER_USAGE_BLAH_BLAH" WTF?` )
3. Or ask the person who wrote the example :D

Important note: you definitely need `EUF_TRANSFER_DST_BIT` because you're copying into this buffer from another one -> If you run the example in DEBUG I put an assert for this in `createFilledDeviceLocalBufferOnDedMem` :)

Reply:
I'm surprised that 2. is a valid way to do that
Is there a 2.5 step for gl?

Reply:
GL doesn't give a shit